File: Leet/283.move_zeros-移动零/283.move_zeros.go
// O(n)
func moveZeroes(nums []int) {
	if nums == nil || len(nums) == 0 {
		return
	}

	left, right, n := 0, 0, len(nums)
	for right < n {
		if nums[right] != 0 {
			nums[left], nums[right] = nums[right], nums[left]
			left++
		}
		right++
	}
}

File: Leet/283.move_zeros-移动零/283.move_zeros.js
// https://leetcode-cn.com/problems/move-zeroes/

// js O(n)
var moveZeroes = function (nums) {
    if (nums == null || nums.length === 0) {
        return;
    }

    let j = 0;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] != 0) {
            nums[j] = nums[i];
            j++;
        }
    }
    for (let i = j; i < nums.length; i++) {
        nums[i] = 0;
    }
};

